src/frontend/src/expr/session_timezone.rs

```diff
@@ -135,7 +135,90 @@ impl SessionTimezone {
                 None
             }
             // TODO: handle tstz-related arithmetic with timezone
-            ExprType::Add | ExprType::Subtract => None,
+            // We first translate to timestamp to handle years, months and days,
+            // then we translate back to timestamptz handle hours and milliseconds
+            //
+            // For performance concern, we assume that all the intervals are const-evaled.
```

Comment:
What about "most intervals could be const-evaled"?

Also to clarify, this const-eval has not been implemented.

Reply:
Yes, it will be very soon, via https://github.com/risingwavelabs/risingwave/pull/7541

Reply:
Cool!